spec/controllers/course/mail_templates_controller_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+RSpec.describe Course::MailTemplatesController, :type => :controller do
+  before do
+    ActsAsTenant.current_tenant = Instance.default
+    sign_in(create(:administrator))
+  end
+  let!(:course) { create(:course) }
+  let!(:existing) { create(:invitation, course: course) }
+  let(:new_template) { build(:announcement, course: course).as_json }
+
+  def expect_correct_response(redirect_path, controller_action, options = {})
+    expect(response).to redirect_to(redirect_path)
+    if options[:mail_action]
+      expect(flash[:notice]).
+        to match(I18n.t("course.mail_templates.#{controller_action}.notice_format") %
+                   { action: I18n.t("course.mail_templates.action.#{options[:mail_action]}") })
+    elsif options[:error]
+      expect(flash[:error]).
+        to match(I18n.t("course.mail_templates.#{controller_action}.error_format") %
+                   { reason: options[:error] })
+    end
+  end
+
+  describe '#index' do
+    subject { get :index, course_id: course.id }
+    it { is_expected.to render_template(:index) }
+  end
+
+  describe '#new' do
+    subject { get :new, course_id: course.id }
+    it { is_expected.to render_template(:new) }
+  end
+
+  describe '#create' do
+    it 'creates a new mail template and redirects to the edit page' do
+      expect { post :create, course_id: course.id, course_mail_template: new_template }.
+        to change(Course::MailTemplate, :count).by(1)
+      created = Course::MailTemplate.all.last
+      expect_correct_response(course_mail_templates_path(course, mail_action: created.action),
+        'create',
+        { mail_action: 'announcement' })
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Redundant curly braces around a hash parameter.